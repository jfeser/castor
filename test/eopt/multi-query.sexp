((queries
  "select([l_shipdate], filter(l_orderkey = param0, lineitem))"
  "join(l_orderkey = o_orderkey, lineitem, orders)")
 (transforms elim-eq-filter elim-join-hash)
 (db postgresql:///tpch_1k))
