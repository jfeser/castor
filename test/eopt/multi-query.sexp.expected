$0:
  lineitem
  alist($0, $32)
  
$1:
  select([l_comment, l_commitdate, l_discount, l_extendedprice, l_linenumber,
          l_linestatus, l_orderkey, l_partkey, l_quantity, l_receiptdate,
          l_returnflag, l_shipdate, l_shipinstruct, l_shipmode, l_suppkey,
          l_tax], $8)
  filter((l_orderkey = param0), $0)
  alist($1, $32)
  
$2:
  select([l_shipdate], $1)
  alist($2, $35)
  
$3:
  orders
  alist($3, $46)
  
$4:
  join((l_orderkey = o_orderkey), $0, $3)
  depjoin($0, $14)
  depjoin($0, $65)
  alist($4, $48)
  
$5:
  select([l_comment, l_commitdate, l_discount, l_extendedprice, l_linenumber,
          l_linestatus, l_orderkey, l_partkey, l_quantity, l_receiptdate,
          l_returnflag, l_shipdate, l_shipinstruct, l_shipmode, l_suppkey,
          l_tax], $67)
  filter((l_orderkey = 0.l_orderkey), $0)
  alist($5, $32)
  
$6:
  select([l_orderkey], $0)
  alist($6, $51)
  
$7:
  dedup($6)
  alist($7, $51)
  
$8:
  alist($8, $54)
  ahashidx($7, $5, param0)
  
$10:
  select([o_clerk, o_comment, o_custkey, o_orderdate, o_orderkey,
          o_orderpriority, o_orderstatus, o_shippriority, o_totalprice], $72)
  filter((0.key = o_orderkey), $3)
  alist($10, $46)
  
$11:
  select([o_orderkey as key], $3)
  alist($11, $58)
  
$12:
  dedup($11)
  alist($12, $58)
  
$13:
  alist($13, $61)
  ahashidx($12, $10, 0.l_orderkey)
  
$14:
  select([0.l_comment, 0.l_commitdate, 0.l_discount, 0.l_extendedprice,
          0.l_linenumber, 0.l_linestatus, 0.l_orderkey, 0.l_partkey,
          0.l_quantity, 0.l_receiptdate, 0.l_returnflag, 0.l_shipdate,
          0.l_shipinstruct, 0.l_shipmode, 0.l_suppkey, 0.l_tax, o_clerk,
          o_comment, o_custkey, o_orderdate, o_orderkey, o_orderpriority,
          o_orderstatus, o_shippriority, o_totalprice], $13)
  alist($14, $48)
  
$16:
  ascalar(0.l_comment)
  alist($16, $76)
  
$17:
  ascalar(0.l_commitdate)
  alist($17, $78)
  
$18:
  ascalar(0.l_discount)
  alist($18, $80)
  
$19:
  ascalar(0.l_extendedprice)
  alist($19, $82)
  
$20:
  ascalar(0.l_linenumber)
  alist($20, $84)
  
$21:
  ascalar(0.l_linestatus)
  alist($21, $86)
  
$22:
  ascalar(0.l_orderkey)
  alist($22, $51)
  
$23:
  ascalar(0.l_partkey)
  alist($23, $89)
  
$24:
  ascalar(0.l_quantity)
  alist($24, $91)
  
$25:
  ascalar(0.l_receiptdate)
  alist($25, $93)
  
$26:
  ascalar(0.l_returnflag)
  alist($26, $95)
  
$27:
  ascalar(0.l_shipdate)
  alist($27, $35)
  
$28:
  ascalar(0.l_shipinstruct)
  alist($28, $98)
  
$29:
  ascalar(0.l_shipmode)
  alist($29, $100)
  
$30:
  ascalar(0.l_suppkey)
  alist($30, $102)
  
$31:
  ascalar(0.l_tax)
  alist($31, $104)
  
$32:
  alist($32, $32)
  atuple([$16, $17, $18, $19, $20, $21, $22, $23, $24, $25, $26, $27, 
          $28, $29, $30, $31], cross)
  
$35:
  alist($35, $35)
  atuple([$27], cross)
  
$37:
  ascalar(0.o_clerk)
  alist($37, $108)
  
$38:
  ascalar(0.o_comment)
  alist($38, $110)
  
$39:
  ascalar(0.o_custkey)
  alist($39, $112)
  
$40:
  ascalar(0.o_orderdate)
  alist($40, $114)
  
$41:
  ascalar(0.o_orderkey)
  alist($41, $116)
  
$42:
  ascalar(0.o_orderpriority)
  alist($42, $118)
  
$43:
  ascalar(0.o_orderstatus)
  alist($43, $120)
  
$44:
  ascalar(0.o_shippriority)
  alist($44, $122)
  
$45:
  ascalar(0.o_totalprice)
  alist($45, $124)
  
$46:
  alist($46, $46)
  atuple([$37, $38, $39, $40, $41, $42, $43, $44, $45], cross)
  
$48:
  alist($48, $48)
  atuple([$16, $17, $18, $19, $20, $21, $22, $23, $24, $25, $26, $27, 
          $28, $29, $30, $31, $37, $38, $39, $40, $41, $42, $43, $44, 
          $45], cross)
  
$51:
  alist($51, $51)
  atuple([$22], cross)
  
$54:
  alist($54, $54)
  atuple([$22, $16, $17, $18, $19, $20, $21, $22, $23, $24, $25, $26, 
          $27, $28, $29, $30, $31], cross)
  
$57:
  ascalar(0.key)
  alist($57, $58)
  
$58:
  alist($58, $58)
  atuple([$57], cross)
  
$61:
  alist($61, $61)
  atuple([$57, $37, $38, $39, $40, $41, $42, $43, $44, $45], cross)
  
$64:
  select([o_clerk, o_comment, o_custkey, o_orderdate, o_orderkey,
          o_orderpriority, o_orderstatus, o_shippriority, o_totalprice], $74)
  filter((0.l_orderkey = o_orderkey), $3)
  alist($64, $46)
  
$65:
  alist($65, $48)
  atuple([$32, $64], cross)
  
$67:
  alist($67, $54)
  ahashidx($7, $5, 0.l_orderkey)
  
$69:
  select([o_clerk, o_comment, o_custkey, o_orderdate, o_orderkey,
          o_orderpriority, o_orderstatus, o_shippriority, o_totalprice], $142)
  filter((o_orderkey = 0.o_orderkey), $3)
  alist($69, $46)
  
$70:
  select([o_orderkey], $3)
  alist($70, $116)
  
$71:
  dedup($70)
  alist($71, $116)
  
$72:
  alist($72, $139)
  ahashidx($71, $69, 0.key)
  
$74:
  alist($74, $139)
  ahashidx($71, $69, 0.l_orderkey)
  
$76:
  alist($76, $76)
  atuple([$16], cross)
  
$78:
  alist($78, $78)
  atuple([$17], cross)
  
$80:
  alist($80, $80)
  atuple([$18], cross)
  
$82:
  alist($82, $82)
  atuple([$19], cross)
  
$84:
  alist($84, $84)
  atuple([$20], cross)
  
$86:
  alist($86, $86)
  atuple([$21], cross)
  
$89:
  alist($89, $89)
  atuple([$23], cross)
  
$91:
  alist($91, $91)
  atuple([$24], cross)
  
$93:
  alist($93, $93)
  atuple([$25], cross)
  
$95:
  alist($95, $95)
  atuple([$26], cross)
  
$98:
  alist($98, $98)
  atuple([$28], cross)
  
$100:
  alist($100, $100)
  atuple([$29], cross)
  
$102:
  alist($102, $102)
  atuple([$30], cross)
  
$104:
  alist($104, $104)
  atuple([$31], cross)
  
$108:
  alist($108, $108)
  atuple([$37], cross)
  
$110:
  alist($110, $110)
  atuple([$38], cross)
  
$112:
  alist($112, $112)
  atuple([$39], cross)
  
$114:
  alist($114, $114)
  atuple([$40], cross)
  
$116:
  alist($116, $116)
  atuple([$41], cross)
  
$118:
  alist($118, $118)
  atuple([$42], cross)
  
$120:
  alist($120, $120)
  atuple([$43], cross)
  
$122:
  alist($122, $122)
  atuple([$44], cross)
  
$124:
  alist($124, $124)
  atuple([$45], cross)
  
$139:
  alist($139, $139)
  atuple([$41, $37, $38, $39, $40, $41, $42, $43, $44, $45], cross)
  
$142:
  alist($142, $139)
  ahashidx($71, $69, 0.o_orderkey)
  
